reflex/utils/prerequisites.py

```diff
@@ -276,6 +267,47 @@ def remove_existing_bun_installation():
         path_ops.rm(os.path.expandvars(constants.BUN_ROOT_PATH))
 
 
+def validate_and_install_node():
+    """Validate nodejs have install or not."""
+    if not check_node_version():
+        install_node()
+
+
+def install_node():
+    """Install nvm and nodejs onto the user's system.
+
+
+    Raises:
+        FileNotFoundError: if unzip or curl packages are not found.
+        Exit: if installation failed
+    """
+    if platform.system() == "Linux":
+        # Only install if bun is not already installed.
+        console.log("Installing nvm...")
+
+        # Check if curl is installed
+        curl_path = path_ops.which("curl")
+        if curl_path is None:
+            raise FileNotFoundError("Reflex requires curl to be installed.")
+
+        result = subprocess.run(constants.INSTALL_NVM, shell=True)
```

Comment:
we should check here if nvm is already installed to avoid duplicate installation.

Reply:
Also, there are some edge cases I think arent being handled here:

- If nvm is already installed, there might be a chance that there could be multiple versions of node installed. We need to check if there's a supported node version(`nvm ls`). If there is, we should switch to that version(`nvm use <version>`). If not, we can go ahead and install the required version and switch to that version(very important as the version after installing still remains the previous version until switched)

- We also need a node version check on `reflex run` as the app will still compile even if the node version does not meet the requirements